src/connectedk8s/azext_connectedk8s/_client_factory.py

```diff
@@ -30,36 +48,72 @@ def cf_connected_cluster_prev_2022_10_01(cli_ctx, _):
 
 def cf_connectedmachine(cli_ctx, subscription_id):
     from azure.mgmt.hybridcompute import HybridComputeManagementClient
+    if os.getenv('AZURE_ACCESS_TOKEN'):
+        credential = AccessTokenCredential(access_token=os.getenv('AZURE_ACCESS_TOKEN'))
+        return get_mgmt_service_client(cli_ctx, HybridComputeManagementClient, subscription_id=subscription_id, credential=credential).private_link_scopes
     return get_mgmt_service_client(cli_ctx, HybridComputeManagementClient, subscription_id=subscription_id).private_link_scopes
 
 
 def cf_resource_groups(cli_ctx, subscription_id=None):
-    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES,
-                                   subscription_id=subscription_id).resource_groups
+    return _resource_client_factory(cli_ctx, subscription_id).resource_groups
 
 
 def _resource_client_factory(cli_ctx, subscription_id=None):
+    from azure.mgmt.resource import ResourceManagementClient
+    if os.getenv('AZURE_ACCESS_TOKEN'):
+        credential = AccessTokenCredential(access_token=os.getenv('AZURE_ACCESS_TOKEN'))
+        return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES, subscription_id=subscription_id, credential=credential)
     return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES, subscription_id=subscription_id)
 
 
-def _resource_providers_client(cli_ctx):
-    from azure.mgmt.resource import ResourceManagementClient
-    return get_mgmt_service_client(cli_ctx, ResourceManagementClient).providers
+def resource_providers_client(cli_ctx, subscription_id=None):
+    return _resource_client_factory(cli_ctx, subscription_id).providers
 
     # Alternate: This should also work
     # subscription_id = get_subscription_id(cli_ctx)
     # return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES, subscription_id=subscription_id).providers
 
 
 def _graph_client_factory(cli_ctx, **_):
-    profile = Profile(cli_ctx=cli_ctx)
-    cred, _, tenant_id = profile.get_login_credentials(
-        resource=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
-    client = GraphRbacManagementClient(cred, tenant_id,
-                                       base_url=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
+    if os.getenv('AZURE_ACCESS_TOKEN'):
+        credential = AccessTokenCredential(access_token=os.getenv('AZURE_ACCESS_TOKEN'))
+        client = GraphRbacManagementClient(credential, os.getenv('AZURE_TENANT_ID'),
```

Comment:
Wait, the same token for ARM won't work for Graph as the resource/scope of the token is incorrect. You will need to provide a separate access token for Graph invocations.

Reply:
That is correct. And that's why the partners who are going to use this feature will not be calling this api. They will be passing the '--custom-location-oid' parameter which will skip the calling of this api. So, its okay for us to merge this change :)